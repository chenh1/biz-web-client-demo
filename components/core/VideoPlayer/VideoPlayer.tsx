'use-client'
import React from "react"
import ReactPlayer from "react-player"

export type VideoPlayerType = {
  isResponsive: boolean;
  width?: `${number}px`;
  height?: `${number}px`;
  url?: string;
  onEnded?: () => void;
  controls?: boolean;
}

export const VideoPlayer: React.FC<VideoPlayerType> = ({ isResponsive, width = '640px', height = '360px', ...props }) => {
  return isResponsive 
  ? <div className="w-full relative pt-[56.25%]">
      <ReactPlayer className={`absolute top-0 left-0`} width={isResponsive ? '100%' : width} height={isResponsive ? '100%' : height} {...props}/>  
    </div>
  : <ReactPlayer {...props}/> 
}