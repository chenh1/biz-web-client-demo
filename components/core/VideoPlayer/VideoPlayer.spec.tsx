import { VideoPlayer } from "./VideoPlayer";
import { render } from '@testing-library/react';
import { JSDOM } from "jsdom"
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window


it('renders responsive react video player', async () => {
  const { asFragment } = render(
    <VideoPlayer isResponsive={true} />
  );

  expect(asFragment()).toMatchSnapshot();
});

it('renders react video player', async () => {
  const { asFragment } = render(
    <VideoPlayer isResponsive={false} />
  );

  expect(asFragment()).toMatchSnapshot();
});
