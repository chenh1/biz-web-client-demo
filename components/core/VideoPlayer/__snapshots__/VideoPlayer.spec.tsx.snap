// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders react video player 1`] = `
<DocumentFragment>
  <div
    style="width: 640px; height: 360px;"
  />
</DocumentFragment>
`;

exports[`renders responsive react video player 1`] = `
<DocumentFragment>
  <div
    class="w-full relative pt-[56.25%]"
  >
    <div
      class="absolute top-0 left-0"
      style="width: 100%; height: 100%;"
    />
  </div>
</DocumentFragment>
`;
